src/main/java/com/epam/aidial/core/controller/PublicationController.java

```diff
@@ -91,22 +92,57 @@ public Future<?> deletePublication() {
                 .compose(body -> {
                     String url = ProxyUtil.convertToObject(body, ResourceLink.class).url();
                     ResourceDescription resource = decodePublication(url);
-                    checkAccess(resource);
+                    checkAccess(resource, false, true);
                     return vertx.executeBlocking(() -> publicationService.deletePublication(resource));
                 })
-                .onSuccess(deleted -> context.respond(deleted ? HttpStatus.OK : HttpStatus.NOT_FOUND))
+                .onSuccess(publication -> context.respond(HttpStatus.OK))
                 .onFailure(error -> respondError("Can't delete publication", error));
 
         return Future.succeededFuture();
     }
 
+    public Future<?> approvePublication() {
+        context.getRequest()
+                .body()
+                .compose(body -> {
+                    String url = ProxyUtil.convertToObject(body, ResourceLink.class).url();
+                    ResourceDescription resource = decodePublication(url);
+                    checkAccess(resource, true, true);
+                    return vertx.executeBlocking(() ->
+                            lockService.underBucketLock(BlobStorageUtil.PUBLIC_LOCATION,
+                                    () -> publicationService.approvePublication(resource)));
+                })
+                .onSuccess(publication -> context.respond(HttpStatus.OK, publication))
+                .onFailure(error -> respondError("Can't approve publication", error));
+
+        return Future.succeededFuture();
+    }
+
+    public Future<?> rejectPublication() {
+        context.getRequest()
+                .body()
+                .compose(body -> {
+                    String url = ProxyUtil.convertToObject(body, ResourceLink.class).url();
+                    ResourceDescription resource = decodePublication(url);
+                    checkAccess(resource, true, true);
+                    return vertx.executeBlocking(() -> publicationService.rejectPublication(resource));
```

Comment:
I think reject also require publication lock

Reply:
Why?

Reply:
I see, you have a double check